src/gpu/optixPrograms.cu

```diff
@@ -108,6 +108,14 @@ __forceinline__ __device__ void saveRayResult(const Vec3f& xyz = Vec3f{NON_HIT_V
 extern "C" __global__ void __raygen__()
 {
 	if (ctx.scene == 0) {
+		if (ctx.doCustomNonHitDistances) {
+			Mat3x4f ray = ctx.rays[optixGetLaunchIndex().x];
+			Vec3f origin = ray * Vec3f{0, 0, 0};
+			Vec3f dir = ray * Vec3f{0, 0, 1} - origin;
+			Vec3f xyz = origin + dir.normalized() * ctx.farNonHitDistance;
+			saveRayResult<false>(xyz, ctx.farNonHitDistance);
+			return;
+		}
```

Comment:
I would think about reducing repetitive code here in `optixPrograms.cu`. We can huddle to discuss it.